src/components/toc.js

```diff
@@ -26,39 +23,41 @@ const Toc = ({ toc }) => {
 const createIntersectionObserver = () => {
 	// gripes with intersection observer: https://twitter.com/nipafx/status/1217590444772810753
 
-	// to make sure only events after the initial observation are acted on,
-	// store the ids of visited headers
-	const visited = []
-	const update = events =>
-		events.forEach(event => {
-			const id = event.target.querySelector(`span`).id
-			if (visited.includes(id)) {
-				// `event.isIntersecting` should indicate whether the element
-				// transitioned into or out of intersection, but Firefox instead
-				// returns true if intersection ratio is > 0;
-				// instead fall back to comparing intersection ratio with threshold
-				const visible = event.intersectionRatio < INTERSECTION_THRESHOLD
-				if (visible) {
-					const position = event.boundingClientRect.top
-					updateHighlight(id, position)
-				}
-			} else visited.push(id)
-		})
-	return new IntersectionObserver(update, {
+	// registering the intersection observer triggers an event for each
+	// observed header, but none of them should be highlighted at this point;
+	// instead store the initial events' targets and only act on an event
+	// if its target was stored earlier.
+	const visited = {}
+
+	const updateHighlights = event => {
+		const focusedHeaderId = event.target.querySelector(`span`).id
+		if (!visited[focusedHeaderId]) {
+			visited[focusedHeaderId] = true
+			return
+		}
+
+		const isHeaderVisible = event.intersectionRatio < INTERSECTION_THRESHOLD
+		if (isHeaderVisible) {
+			// `event.isIntersecting` should indicate whether the element
+			// transitioned into or out of intersection, but Firefox instead
+			// returns true if intersection ratio is > 0;
+			// instead fall back to comparing intersection ratio with threshold
```

Comment:
My initial "dare" to refactor was my understanding that you objected to this comment (and others) and wanted to express it with cleaner code. Your refactoring removed the comment, but the resulting code does not explain its content.

This comment addresses developers who are aware of the intersection observer API (e.g. future me) and might wonder why I'm making this more complicated than it has to be and don't just use `event.isIntersecting` when it's clearly meant to provide the exact information I'm looking for here.